src/main/java/io/anserini/search/topicreader/TsvStringKeyTopicReader.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Anserini: A Lucene toolkit for replicable information retrieval research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.search.topicreader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+/**
+ * Topic reader for queries in tsv format, such as the MS MARCO queries.
+ *
+ * <pre>
+ * 174249 does xpress bet charge to deposit money in your account
+ * 320792 how much is a cost to run disneyland
+ * 1090270  botulinum definition
+ * 1101279	do physicians pay for insurance from their salaries?
+ * 201376 here there be dragons comic
+ * 54544  blood diseases that are sexually transmitted
+ * ...
+ * </pre>
+ */
+public class TsvStringKeyTopicReader extends TopicReader<String> {
```

Comment:
This is mostly duplicate with https://github.com/castorini/anserini/blob/master/src/main/java/io/anserini/search/topicreader/TsvTopicReader.java

Seems like the only different is either treating the key as an int or the key as a string...

I don't know if the distinction is important, but let's keep it for now.

So, refactor: `TsvTopicReader` -> `TsvIntTopicReader`, this class to `TsvStringTopicReader`?

Reply:
Any reason we want to use int instead of string?

Reply:
I think it's the sort order? 100 sorts before 99 based on String.

Actually, you know what? Refactor into a link list map, and let's just preserve whatever order comes in the topic file.

That way we don't need two separate implementations.

The key can be String in this case.

What do you think?